RecoTracker/TkHitPairs/plugins/HitPairEDProducer.cc

```diff
@@ -504,6 +519,11 @@ void HitPairEDProducer::fillDescriptions(edm::ConfigurationDescriptions& descrip
   desc.add<bool>("produceIntermediateHitDoublets", false);
   desc.add<unsigned int>("maxElement", 1000000);
   desc.add<unsigned int>("maxElementTotal", 50000000);
+  desc.add<bool>("putEmptyIfMaxElementReached", true)
```

Comment:
The default should likely be `false` and enabled only in specific modules in the offline reconstruction.

Reply:
I think that the default logic should stay: we make doublets where we can until the total maximum.
Even in this specific case of `pixelPairElectronHitDoublets`, if e.g. `FPix1_pos+FPix3_pos` goes above the max, 1+2 and 2+3 should still be tried and tracking will be meaningful.

How does the RSS use scale with the max values here in terms of bytes/element?
 would a more appropriate solution be to tighten the `maxElementTotal` ?

Reply:
> Even in this specific case of `pixelPairElectronHitDoublets`, if e.g. `FPix1_pos+FPix3_pos` goes above the max, 1+2 and 2+3 should still be tried and tracking will be meaningful.

Ok. Just to check I understood correctly, are you implying that the current behavior is preferred and this PR could be closed?

> How does the RSS use scale with the max values here in terms of bytes/element?

I have not checked.

> would a more appropriate solution be to tighten the `maxElementTotal` ?

Perhaps, I have not checked.

Reply:
> Ok. Just to check I understood correctly, are you implying that the current behavior is preferred and this PR could be closed?

I don't have a significant objection against increased configurability and could see a case where all pairs/sets are essential.
So, the PR could stay, but the default here better be `false`.

Reply:
Default changed, "RFC" taken, out, and master (https://github.com/cms-sw/cmssw/pull/41561) and 13_1_X (https://github.com/cms-sw/cmssw/pull/41562) PRs opened.